Resolve #27; resolve #15; resolve #26

elsayeaa opened the pull request:
I worked with @nidhi2509 on writing a README for the current project, and we are waiting for feedback.

ProfJanetDavis committed:
Created tests for the Header component
As part of this, we set up mocha, chai, and puppeteer.
We loosely followed the tutorial at
https://medium.com/@ankit_m/ui-testing-with-puppeteer-and-mocha-part-1-getting-started-b141b2f9e21

ProfJanetDavis committed:
Modified yarn serve to run at localhost + 4-sp tabs

ProfJanetDavis committed:
Fixed spacing

ProfJanetDavis committed:
Refer to "app title" instead of "app name"

ProfJanetDavis committed:
Outline tests for the Readout component

dylanjwu committed:
Create test for project url in footer

ProfJanetDavis committed:
Replace autogenerated README w/ meaningful content
This is the work of @elsayeaa and @Nidhi

ProfJanetDavis commented on README.md:
```diff
@@ -1,24 +1,92 @@
+# Reading For Gender Bias - FrontEnd (GEL team)
+​
+## Table of contents 
+​
```
Please delete extra blank lines between headers and list content.

ProfJanetDavis commented on README.md:
```diff
@@ -1,24 +1,92 @@
+​
+## Introduction
+​
+This project is a user-interface web application that employs *Reading for Gender Bias* back-end, [more information here](https://github.com/glam-lab/gender-bias) to create an accessible, text-analysis tool that directs recommenders and evaluators to avoid implicit gender biases in their writings. This process is done through flagging gendered words, phrases, and grindstone adjectives, pointing o ...
```
I would make this paragraph quite a bit briefer: "This project is a web-based user interface for [Reading for Gender Bias](url)."

If you think it's important to explain more about what the _Reading for Gender Bias_ project is, I would encourage you to quote the `README` for that project rather than trying to write something original that might not accurately portray @molliem's  intentions. Does that make sense?

ProfJanetDavis replied:
(I'll invite @molliem and @j6k4m8 to review after we've gone through a round or two of revision.)

ProfJanetDavis commented on README.md:
```diff
@@ -1,24 +1,92 @@
+​
+This project is a user-interface web application that employs *Reading for Gender Bias* back-end, [more information here](https://github.com/glam-lab/gender-bias) to create an accessible, text-analysis tool that directs recommenders and evaluators to avoid implicit gender biases in their writings. This process is done through flagging gendered words, phrases, and grindstone adjectives, pointing o ...
+​
+This project is currently a prototype that is under modification process to be turned into a usable product. Current modifications are monitored through Whitman College Computer Science Summer Research team.
```
"This project is currently under active development by members of the [GLAM Lab](https://github.com/glam-lab) at Whitman College."

ProfJanetDavis commented on README.md:
```diff
@@ -1,24 +1,92 @@
+​
+## Technologies
+​
+* Javascript
```
I don't think you need to mention Javascript, HTML, or CSS. Those are all a given for any front-end web app, and they are all implied by our adoption of the Vue framework.

ProfJanetDavis commented on README.md:
```diff
@@ -1,24 +1,92 @@
+* Puppeteer, mocha, chai for testing
+* HTML
+* CSS
+* yarn/npm as package managers
```
This bullet list is kind of awkward. I'd encourage you to think about whether you could write a short paragraph instead.

Also, please add links to the home pages for these frameworks and tools. This is a good documentation practice - super helpful for new developers (including ourselves).

ProfJanetDavis commented on README.md:
````diff
@@ -1,24 +1,92 @@
+``` 
+### Running the front-end 
+​
+Once installed, add install Vue, and use yarn to open the server locally. Go to the right directory, and start the server. 
````
I don't think this sentence adds anything to the commands below, so I would delete it. 

Software developers are impatient people. Concision and simplicity are great virtues in technical writing.

ProfJanetDavis commented on README.md:
````diff
@@ -1,24 +1,92 @@
 ```
+​
+​
+## Features
````
This section is going to be hard to keep up to date. Delete it.

ProfJanetDavis commented on README.md:
```diff
@@ -1,24 +1,92 @@
+## Limitations
+* Plug-ins are not available, which requires the recommender/evaluator to visit the currently developed website. 
+* Naturally processing recommendation letters and using them for further research regarding implicit gender biases. 
+* There is no way to export the final document as a PDF/docx/doc etc.
```
I would delete this section too.

ProfJanetDavis commented on README.md:
```diff
@@ -1,24 +1,92 @@
-See [Configuration Reference](https://cli.vuejs.org/config/).
+## Status
+​
+This project is currently being developed to be more accessible for users and developers/contributors. 
```
Delete this. You already wrote about the project status in the introduction.

ProfJanetDavis commented on README.md:
```diff
@@ -1,24 +1,92 @@
+​
+## Resources
+​
+[Reading for Gender Bias Repistory](https://github.com/glam-lab/gender-bias#reading-for-gender-bias) 
```
Delete this section too. All important resources have already been cited above.

ProfJanetDavis commented on README.md:
```diff
@@ -1,24 +1,92 @@
+​
+## Credits
+​
+Credits are given to @Mollie, @Jordan for their contributions as product owners.
```
- This should be titled "Contributors."
- All the members of our team should be listed as the main developers for Summer 2020.
- Mollie for conceiving the project and guiding our development.
- Jordan should be thanked for starting the project and for his mentorship.
- Make sure you are using the correct @ names for everyone.

ProfJanetDavis reviewed (changes requested):
Hi Ahmed (and Nidhi), this is off to a really good start. It was so helpful for me to have a draft to react to. While I've asked for many changes, I don't think any of them will be too difficult to accomplish. In particular, note several comments are requests for deletion.

ProfJanetDavis committed:
Outline tests for the Blurb component

nidhi2509 committed:
github link fixed

nidhi2509 committed:
bugfix for github link

dylanjwu committed:
set timeout to 100000 in package.json

ProfJanetDavis committed:
Reduced timeout from 100s to 5s

ProfJanetDavis committed:
Test the main interaction with the Readout component:
- Find the textarea and submit button;
- Enter text;
- Click the submit button;
- See a result.

ProfJanetDavis committed:
Define SEL_ consts in boostrap.js

ProfJanetDavis committed:
Set up for testing blurbs

nidhi2509 committed:
blurb tests

dylanjwu committed:
wrote tests 2,4,6,8 for blurb and most of readout

dylanjwu committed:
Delete unnecessary test in readout

nidhi2509 committed:
blurb

nidhi2509 committed:
blurb

nidhi2509 committed:
blurb tests

Life1999 committed:
eslint changes

Life1999 committed:
Merge branch 'automated_testing' of https://github.com/glam-lab/reading-for-genderbias-web into automated_testing

nidhi2509 committed:
added expect statements

Life1999 committed:
Co-authored-by: Dylan Wu <dylanjwu@gmail.com>

Life1999 committed:
fixes issue #15

Life1999 committed:
Merge branch 'automated_testing' of https://github.com/glam-lab/reading-for-genderbias-web into automated_testing

dylanjwu committed:
Refactor to be more consistent from PR review

dylanjwu committed:
Merge README test suite guide from other branch

dylanjwu committed:
Merge branch 'master' into automated_testing

dylanjwu committed:
Fix small mistake in package.json

dylanjwu committed:
mend

dylanjwu committed:
Merge pull request #2 from glam-lab/automated_testing

Automated testing

dylanjwu committed:
Merge branch 'master' into fix-issue_26

dylanjwu committed:
Set up fake server to run when app starts

dylanjwu committed:
mend

dylanjwu committed:
Enable and documented fake server and global TEXT

dylanjwu committed:
Enable fake server and global TEXT for testing

dylanjwu committed:
Merge branch 'fix-issue_26' of https://github.com/glam-lab/reading-for-genderbias-web into fix-issue_26

elsayeaa committed:
Update README.md

elsayeaa committed:
Update README.md

elsayeaa committed:
Update README.md

ProfJanetDavis committed:
Format introduction as prose, not code

ProfJanetDavis committed:
Fix Markdown formatting problems

ProfJanetDavis committed:
Revised with @elsayeaa

ProfJanetDavis committed:
Revise for clarity and readability with @elsayeaa

j6k4m8 commented on README.md:
````diff
@@ -1,24 +1,58 @@
 
-### Compiles and minifies for production
 ```
+yarn add Vue / npm install vue
````
The right way to install vue is listed [in the docs](https://cli.vuejs.org/guide/installation.html) as the following:

```shell
npm install -g @vue/cli
# OR
yarn global add @vue/cli
```


...but I don't think this step is necessary anyway, if the user has already run `[yarn|npm] install`.

ProfJanetDavis replied:
I think @j6k4m8 is saying to delete this instruction, because it's unnecessary. Since `vue` is listed in the `yarn.lock` file, it should be installed automatically when you run `yarn install`.

This makes me wonder if we should delete references to `npm` and just recommend `yarn`. Thoughts, @j6k4m8 ?

j6k4m8 replied:
Either works for me. I usually tend to support both but I don't have preferences here.

ProfJanetDavis replied:
We weren't really supporting `npm` so I removed the references.

j6k4m8 commented on README.md:
````diff
@@ -1,24 +1,58 @@
-### Compiles and minifies for production
 ```
+yarn add Vue / npm install vue
+yarn serve /npm run server
````
this should be `npm run serve`, right? (rather than `server`)

```suggestion
yarn serve
# OR
npm run serve
```

ProfJanetDavis replied:
If we delete references to `npm`, this problem goes away.

j6k4m8 commented on README.md:
````diff
@@ -1,24 +1,58 @@
+*To lint and fix files*:
 ```
 yarn lint
 ```
````
Perhaps after this section, add a `### Running Tests` section that explains how to run `yarn serve` and `yarn test` in separate processes?

ProfJanetDavis replied:
Yes, that sounds good unless @dylanjwu and @Life1999 did it already!

ProfJanetDavis replied:
Yes, they did. I just changed the title from "Testing" to "Running tests"

j6k4m8 reviewed (changes requested):
Just a few documentation tweaks :)

All tests pass for me (nice work!!). This is great!

Life1999 committed:
Updated readme and main.js to reflect pr comments

ProfJanetDavis committed:
Merge branch 'master' into fix-issue_26

ProfJanetDavis committed:
Merge pull request #3 from glam-lab/fix-issue_26

Fix issue #26

ProfJanetDavis committed:
"Testing" -> "Running tests" 

Per Jordan's suggestion

ProfJanetDavis committed:
Use yarn only and more consistent style

ProfJanetDavis committed:
Merge branch 'master' into README-version-2

ProfJanetDavis committed:
Merge pull request #4 from glam-lab/README-version-2

Update README.md

ProfJanetDavis committed:
Change the title

ProfJanetDavis committed:
We are committed to yarn.
